Python/Product/Debugger/Debugger/PythonProcess.cs

```diff
@@ -667,10 +667,13 @@ public PythonProcess(PythonLanguageVersion languageVersion, string exe, string a
         private void HandleExecutionException(Stream stream) {
             int execId = stream.ReadInt32();
             CompletionInfo completion;
-
             lock (_pendingExecutes) {
-                completion = _pendingExecutes[execId];
-                _pendingExecutes.Remove(execId);
+                if (_pendingExecutes.TryGetValue(execId, out completion)) {
+                    _pendingExecutes.Remove(execId);
+                    _ids.Free(execId);
+                } else {
+                    Debug.Fail("Received REPL execution result with unknown execution ID " + execId);
+                }
             }
 
             string exceptionText = stream.ReadString();

```

Comment:
There shouldn't be any case where we can successfully retrieve a null completion from the dictionary, since we don't put any there in the first place.